### ISSUE TITLE
[FEATURE] Detect mock location

### ISSUE THREAD

**Issue by kaynz** [dpa99c/cordova-diagnostic-plugin#151]
Would be cool to detect if mock location is enabled:

```
if (Secure.getString(this.cordova.getActivity().getContentResolver(), 
Secure.ALLOW_MOCK_LOCATION).equals("0")) {
	callbackContext.success("false");
		} else {
	callbackContext.success("true");
		}
```

**Comment by dpa99c** [comment]
According to [Android docs](https://developer.android.com/reference/android/provider/Settings.Secure.html#ALLOW_MOCK_LOCATION), mock location setting was deprecated in Marshmallow but should be able to something based on [this SO answer](http://stackoverflow.com/questions/33003553/how-to-read-selected-mock-location-app-in-android-m-api-23/33066797#33066797)

**Comment by jacobg** [comment]
It would be great to add this as a feature, since the web geolocation api doesn't include that information. The code sample above is not a reliable way to detect mocking. There are a lot of false positives, e.g., some stock Samsung apps have that permission.

The best way is to actually collect location data, and call `isFromMockProvider()` on the object:
https://developer.android.com/reference/android/location/Location.html#isFromMockProvider()
